### PR TITLE
Send Otel traces to the control plane.

### PR DESCRIPTION
**Pull request by Warashi**
**What this PR does / why we need it**:

- Add trace spans for the plan and deployment.
- Add default attributes for tracing.
- Disable opentelemetry-collector routes when monitoring isn't enabled.

This is the sample trace sent to the local Jaeger by this implementation.
the sent traces to the local Jaeger

**Which issue(s) this PR fixes**:

Part of #4977

The rest are below.
- add documents.
- add the component to visualize traces.

**Does this PR introduce a user-facing change?**: No

- **How are users affected by this change**:
- **Is this breaking change**:
- **How to migrate (if breaking change)**:

### CODE REVIEW

**Comment by Warashi** [comment]
I found some needs for refactoring, so I changed this PR's state to a draft.

File: pkg/app/piped/controller/scheduler.go [hunk 1]
```diff
@@ -264,6 +270,25 @@ func (s *scheduler) Run(ctx context.Context) error {
...
+	defer span.End()
+
+	defer func() {
+		switch deploymentStatus {
+		case model.DeploymentStatus_DEPLOYMENT_SUCCESS:
+			span.SetStatus(codes.Ok, statusReason)
+		case model.DeploymentStatus_DEPLOYMENT_FAILURE, model.DeploymentStatus_DEPLOYMENT_CANCELLED:
+			span.SetStatus(codes.Error, statusReason)
+		}
+	}()
```
[the remark this answers is not in the document]
Comment (Warashi) [reply pkg/app/piped/controller/scheduler.go]: Thank you. commited suggestion.

**Review by ffjlabo — APPROVED** [review]
LGTM!

**Review by khanhtc1202 — APPROVED** [review]
LGTM 👍

**Pull request merged**